From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 31 May 2021 07:25:58 +0800
Subject: HACK: media: capture: linux: Allow camera without supported format

The chromium would only accept discrete frame sizes.

Hack it to make Rockchip ISP camera working.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>

diff --git a/media/capture/video/linux/video_capture_device_factory_v4l2.cc b/media/capture/video/linux/video_capture_device_factory_v4l2.cc
index ca36ff01b9174d06c1667a78e8792b2679fa4c5d..04349acdaf22a7367eaf4cc8671b21217c92a87b 100644
--- a/media/capture/video/linux/video_capture_device_factory_v4l2.cc
+++ b/media/capture/video/linux/video_capture_device_factory_v4l2.cc
@@ -198,10 +198,8 @@ void VideoCaptureDeviceFactoryV4L2::GetDevicesInfo(
 
       VideoCaptureFormats supported_formats;
       GetSupportedFormatsForV4L2BufferType(fd.get(), &supported_formats);
-      if (supported_formats.empty()) {
-        DVLOG(1) << "No supported formats: " << unique_id;
-        continue;
-      }
+      if (supported_formats.empty())
+        LOG(WARNING) << "No supported formats: " << unique_id;
 
 
       // HACK: Somehow the newest chromium would prefer using the last device in some cases, e.g. apprtc
